app/models/user.rb

```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  validates :email, uniqueness: true, presence: true, format: { with: URI::MailTo::EMAIL_REGEXP }
+  validates_presence_of :password, require: true
+  validates :api_key, uniqueness: true, presence: true
+  has_secure_password
+
+  after_validation(on: :create) do
```

Comment:
For some reason before_create: do would NOT work for me. luckily after_validation(on: :create) fixed the problem